### DISCUSSION TITLE
Rule for named parameters

### DISCUSSION THREAD

**Discussion by remcomokveld** [detekt/detekt#6201] [Ideas]
It is common practice to put the [primary or most important Composable function parameter](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#compose-ui-layouts) as last parameter to ensure that you can specify it as a trailing lambda.

For non-compasable lambda parameters like event handlers, enforcing named can often make the code much more easier to read.

For example

```kotlin
@Composable
fun MyScreen(onItemClicked: (Item) -> Unit) {
    
}


// non-compliant
@Composable
fun ParentComposable() {
  MyScreen { doSomething() }
}

// vs.

// compliant
@Composable
fun ParentComposable() {
  MyScreen(onItemClicked = { doSomething() })
}
```

Since there may also be valid use not using named parameters, I wonder if this could also be done using an opt-in on the declaration side (guessing that makes implementing the rule harder because you'd need to add a detekt dependency to the compile classpath.

Something like
```kotlin
@Composable
fun MyScreen(@RequireNamed onItemClicked: (Item) -> Unit) {
    
}
```

**Comment by schalkms** [comment]
I agree that it's technically possible.
You could take a look at the [IgnoredReturnValue](https://github.com/detekt/detekt/blob/main/detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt) rule. On first glance, it provides similar functionality that you want to achieve.
Maybe, you also need to dig into the Kotlin compiler diagnostics and see how it's done there.
